### ISSUE TITLE
Wrong permission on the user runtime directory

### ISSUE THREAD

**Issue by grulja** [flatpak/flatpak#3397]
## Linux distribution and version
Fedora 31

## Flatpak version
flatpak-1.6.1-2.fc31.x86_64

## Description of the problem
With a recent change in Qt, there is an additional check for correct permission on the user runtime directory (/run/user/xxxx), where when the permission is wrong, we will never get the runtime directory from **QStandardPaths::RuntimeLocation**. This will fail all our detections of flatpak, where we check for flatpak-info file. This also breaks some other stuff and some applications like Dolphin became unusable.

The error from Qt is:
`QStandardPaths: wrong permissions on runtime directory /run/user/1000, 7755 instead of 7700`

Qt upstream change: https://codereview.qt-project.org/c/qt/qtbase/+/256521

A temporary solution would be to revert this change in the KDE runtime so users can continue using KDE/Qt apps, but it should be fixed in Flatpak.

**Comment by ymshenyu** [comment]
It seems like qt 5.14.1 uploaded to the wrong branch. And It also breaks ibus in telegram.

**Comment by Erick555** [comment]
@ymshenyu why wrong branch?

**Comment by ymshenyu** [comment]
@Erick555 They didn't change the branch in the [manifest].(https://github.com/KDE/flatpak-kde-runtime/blob/qt5.14/org.kde.Sdk.json.in)

**Comment by ymshenyu** [comment]
@Erick555  
![image](https://user-images.githubusercontent.com/39402395/73592491-128cf680-4536-11ea-8dab-eb9193644924.png)
I should get qt 5.13.2 instead of 5.14.1. But it seems like qt 5.14.1 uploaded to 5.13 branch.

**Comment by Erick555** [comment]
@ymshenyu I just installed org.kde.Sdk//5.13 from flathubk  and can't reproduce:
```
flatpak run org.kde.Sdk//5.13
[📦 org.kde.Sdk ~]$ qmake --version
QMake version 3.1
Using Qt version 5.13.2 in /usr/lib/x86_64-linux-gnu
[📦 org.kde.Sdk ~]$ 
```
It's pissible that some mistake was done before but is fixed now, could you try `flatpak update`?

**Comment by ymshenyu** [comment]
@Erick555 Thanks for the heads up. But I mean this bug should fix in 5.14.1, not 5.13.2.

**Comment by Erick555** [comment]
So I confirmed that commit for kde org.kde.Platform`56180094df53ad85ae2d3f51cc727337d294e1917e1fe741de494b089f38b950` contains qt 5.14.

**Comment by grulja** [comment]
Ufff, thanks for spotting this, I spent quite a lot of time trying to figure out what's wrong. It should be fixed now, both versions are being currently build.

Still this remains an issue for Qt 5.14, however it doesn't seem to be that serious as I don't expect many apps using it at this moment.

**Comment by Erick555** [comment]
@grulja after 5.14 will be published then apps will start using it so reverting that change for 5.14 sounds right: https://invent.kde.org/kde/flatpak-kde-runtime/commit/734e1b0a31159012f17e59597d3f1039ebb97bc5

**Comment by grulja** [comment]
I reverted that change now in KDE Runtime 5.14 so apps switching to Qt 5.14 won't break until this is fixed.

**Comment by Erick555** [comment]
@grulja bubblewrap always create dirs with 755 permissions so this should be reported there, could you open issue at https://github.com/containers/bubblewrap/issues ?

**Comment by grulja** [comment]
> @grulja bubblewrap always create dirs with 755 permissions so this should be reported there, could you open issue at https://github.com/containers/bubblewrap/issues ?

Opened https://github.com/containers/bubblewrap/issues/346. Thanks.

**Comment by Erick555** [comment]
@grulja the solution now exist on bwrap side (although not in a stable release) so you may re-open this one as the ball is on flatpak side.

**Comment by smcv** [comment]
With bubblewrap updated to a version that contains containers/bubblewrap#406, `--dir /run/user/1000 --chmod 0700 /run/user/1000` (or `--perms 0700 --dir /run/user/1000` if we make sure the directory doesn't already exist) would resolve this.

We shouldn't do that until there is a bubblewrap *release* with those new options, because distros often build flatpak and bubblewrap separately, so there needs to be a version number we can check for in `configure.ac`.

**Comment by smcv** [comment]
90% of this was accidentally fixed in 1.11.1 (specifically commit 38eac072) while resolving #4120. Most app instances now share a per-app-ID `XDG_RUNTIME_DIR` created as `$XDG_RUNTIME_DIR/.flatpak/$FLATPAK_ID/xdg-run` on the host system, and those have the canonical 0700 permissions.

The part that remains to be fixed is that `flatpak-spawn --sandbox` creates a sub-sandbox that does not participate in the per-app-ID `XDG_RUNTIME_DIR` sharing, and processes in those sub-sandboxes still get the 0755 permissions described in this issue.
